client/state/happychat/middleware.js

```diff
@@ -18,6 +25,16 @@ export const requestTranscript = ( connection, { getState, dispatch } ) => {
 	);
 };
 
+export const sendEventMessage = ( connection, { getState }, action ) =>{
+	const state = getState();
+	const currentUser = getCurrentUser( state );
+	if ( isHappychatClientConnected( state ) &&
+		isHappychatAcceptingChats( state ) &&
```

Comment:
I'm not super familiar with all the states of HC yet, but just wanted to sanity check that `isHappychatAcceptingChats` is needed here? In my mind, it seems if you're mid-chat and all operators go Red, `isHappychatAcceptingChats` should turn to `false` — which would mean that you would stop sending route data to your ongoing session. Won't `isHappychatChatAssigned` be a clear enough indicator whether route data should be sent?

If the logic makes sense in a way that I don't fully understand, no worries — just double-checking.